### PR TITLE
Feat graphql

### PR DESCRIPTION
**Pull request by abnegate**
## What does this PR do?

- Add `x-sdk-graphql` header for methods with `graphql` method type, to allow Appwrite to extract parameters from a parameterless route
- Allow list _or_ object responses for strongly typed languages that did not already (Android/Kotlin/Swift/Apple), for GraphQL batched requests which return a list of objects
- Add GraphQL language
- Add GraphQL example template
- Fix REST example template
- Add abstract base class HTTP which REST and GraphQL extend

## Test Plan

Existing tests

## Related PRs and Issues

https://github.com/appwrite/docs/pull/287
https://github.com/appwrite/homepage/pull/113
https://github.com/appwrite/appwrite/pull/974

### Have you read the [Contributing Guidelines on issues](https://github.com/appwrite/appwrite/blob/master/CONTRIBUTING.md)?

Yes

### CODE REVIEW

File: templates/swift/Sources/Client.swift.twig [hunk 1]
```diff
@@ -232,7 +232,7 @@ open class Client {
         headers: [String: String] = [:],
         params: [String: Any?] = [:],
         sink: ((ByteBuffer) -> Void)? = nil,
-        converter: (([String: Any]) -> T)? = nil
+        converter: ((Any) -> T)? = nil
```
Comment (lohanidamodar) [inline templates/swift/Sources/Client.swift.twig:235]: Why was this change necessary on Kotlin and Swift?

**Review by lohanidamodar — APPROVED** [review]

File: templates/http/docs/example.md.twig [hunk 2]
```diff
@@ -1,4 +1,4 @@
-{% set boundary = random() %}
+{% set boundary = 'cec8e8123c05ba25' %}
```
Comment (eldadfux) [inline templates/http/docs/example.md.twig:1]: If the class name is REST, why do we have the templates in HTML directory?

Comment (abnegate) [reply templates/http/docs/example.md.twig:1]: Have moved to a rest directory 👌

**Review by eldadfux — APPROVED** [review]
